### PR TITLE
Deleting form item

### PR DESCRIPTION
**Pull request by manuelcheveste**
Trello link: 
[As an admin I should be able to delete a form](https://trello.com/c/t0m6PvLB/12-3-as-an-admin-i-should-be-able-to-delete-a-form)

- [x] Display a list of forms
- [x] Add delete icon to each form item
- [x] Delete form when the icon is clicked

output

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -13,7 +15,35 @@ const fetchFormCategories = () => ({
...
+    types: [FORMS_REQUEST, FORMS_SUCCESS, FORMS_FAILURE],
+    endpoint: `form_categories/${category}/forms`,
+    schema: Schemas.FORM_ARRAY,
+    refetch
+  }
+});
+
+export const loadForms = (category, refetch) => (dispatch, getState) => {
+  const { forms } = getState().entities;
+  if (Object.keys(forms).length && !refetch) {
```
Comment (mcousillas6) [inline src/actions/index.js]: why object.keys? forms should be an array, right?

File: src/actions/index.js [hunk 2]
```diff
@@ -13,7 +15,35 @@ const fetchFormCategories = () => ({
   }
 });
 
-export const loadFormCategories = () => dispatch => dispatch(fetchFormCategories());
+export const loadFormCategories = () => (dispatch, getState) => {
+  const { formCategories } = getState().entities;
+  if (Object.keys(formCategories).length) {
```
Comment (mcousillas6) [inline src/actions/index.js]: same as with forms, why object.keys?

File: src/actions/index.js [hunk 3]
```diff
@@ -13,7 +15,35 @@ const fetchFormCategories = () => ({
   }
 });
 
-export const loadFormCategories = () => dispatch => dispatch(fetchFormCategories());
+export const loadFormCategories = () => (dispatch, getState) => {
```
Comment (mcousillas6) [inline src/actions/index.js]: I think you need to start splitting your actions into separate files, usually it's a good pattern to have one action file per entity, so categoriesActions and formActions for example, to keeps things organized

File: src/actions/index.js [hunk 4]
```diff
@@ -13,7 +15,35 @@ const fetchFormCategories = () => ({
...
+  const { formCategories } = getState().entities;
+  if (Object.keys(formCategories).length) {
+    return null;
+  }
+  dispatch(fetchFormCategories());
+};
```
Comment (ConstanzaCurzio) [inline src/actions/index.js]: Maybe
```
if (!Object.keys(formCategories).length) {
   dispatch(fetchFormCategories());
}  
```

File: src/actions/index.js [hunk 5]
```diff
@@ -13,7 +15,35 @@ const fetchFormCategories = () => ({
...
+    endpoint: `form_categories/${category}/forms`,
+    schema: Schemas.FORM_ARRAY,
+    refetch
+  }
+});
+
+export const loadForms = (category, refetch) => (dispatch, getState) => {
+  const { forms } = getState().entities;
+  if (Object.keys(forms).length && !refetch) {
+    return null;
```
Comment (ConstanzaCurzio) [inline src/actions/index.js]: Same here, returning null seems odd. Maybe we can restructure this if statement so we don't have to.

File: src/actions/index.js (hunk 1 above)
Comment (manuelcheveste) [reply src/actions/index.js]: I started using normalizr, so entities look something like this
```
entities: {
    formCategories: {
      '1': {
        name: 'Payments',
        version: 1,
        id: 1,
        updatedAt: '2019-10-24T19:16:03.860Z',
        createdAt: '2019-10-24T19:16:03.860Z'
      },
      ...
}
```

File: src/actions/index.js (hunk 5 above)
Comment (manuelcheveste) [reply src/actions/index.js]: Will do like the comment above

**Review by ConstanzaCurzio — APPROVED** [review]

**Review by mcousillas6 — APPROVED** [review]

**Review by xlasserre — APPROVED** [review]

File: src/actions/formActions.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+export const FORMS_REQUEST = 'FORMS_REQUEST';
+export const FORMS_SUCCESS = 'FORMS_SUCCESS';
+export const FORMS_FAILURE = 'FORMS_FAILURE';
+
+const fetchForms = (category, refetch) => ({
+  category,
+  [CALL_API]: {
+    types: [FORMS_REQUEST, FORMS_SUCCESS, FORMS_FAILURE],
+    endpoint: `form_categories/${category}/forms`,
```
Comment (agustito37) [inline src/actions/formActions.js:14]: should the endpoint's url be defined in here?

**Review by agustito37 — APPROVED** [review]

File: src/actions/formActions.js (hunk 6 above)
Comment (manuelcheveste) [reply src/actions/formActions.js:14]: I don't see it necessary to extract it yet because currently, only the action needs to know which endpoint to use

**Pull request merged**